lib/bot.js

```diff
@@ -36,6 +36,8 @@ class Bot {
     this.channels = _.values(options.channelMapping);
     this.ircStatusNotices = options.ircStatusNotices;
     this.announceSelfJoin = options.announceSelfJoin;
+    this.ignoreBots = options.ignoreBots;
+    this.ignoreIrcNicks = options.ignoreIrcNicks || [];
```

Comment:
Well it depends for what the ignore would be used the most, if it's used to prevent some bots to go through username#separator would be fine. But if it's to prevent some user to flood and kicking/banning/mute isn't option for some reason, he can just switch discord username to circumvent the ignore. 

In both cases I don't think the getting userid part would be too complicated, most people running servers and bots already have dev mode on all the time.